src/Utils/Utils.php

```diff
@@ -442,7 +442,7 @@ public static function printSafe($var)
      */
     public static function chr($ord, $encoding = 'UTF-8')
     {
-        if ($ord <= 255) {
+        if ($ord <= 127) {
```

Comment:
I guess this check only makes sense when $encoding is `UTF-8`. But we can even try deleting this check altogether since it was a performance optimization that is not required anymore as far as I remember.

Reply:
I pushed another commit removing the check entirely per your feedback.

The problem I mentioned about characters like "𝕄𝕖𝕥𝕒𝕝𝕝𝕚𝕔" are a separate problem (UTF-16 surrogates aren't handled in the lexer). I'll mail a patch for that early next week